UnityProject/Assets/Scripts/Systems/Antagonists/Antags/Changeling/ChangelingMain.cs

```diff
@@ -0,0 +1,626 @@
+using Mirror;
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using Systems.Character;
+using UnityEngine;
+using Util;
+
+namespace Changeling
+{
+	[DisallowMultipleComponent]
+	public class ChangelingMain : NetworkBehaviour
+	{
+		[Header("Changeling main")]
+		[SyncVar(hook = nameof(SyncChemCount))]
+		private float chem = 25;
+		public float Chem => chem;
+		[SyncVar(hook = nameof(SyncMindID))] private uint changelingMindID = 0;
+		[SyncVar] private float chemMax = 75;
+		[SyncVar] private float chemAddPerTime = CHEM_ADD_TIME;
+		[SyncVar] private float chemAddTime = CHEM_ADD_PER_TIME_BASE;
+		public int ExtractedDna => changelingDnas.Count;
+
+		[SyncVar(hook = nameof(SyncResetCounts))] private int resetCount = 0;
+		[SyncVar(hook = nameof(SyncMaxResetCounts))] private int resetCountMax = 1;
+
+		[SyncVar] private int absorbCount = 0;
+		public int AbsorbCount => absorbCount;
+		public int ResetsLeft => resetCountMax - resetCount;
+
+		private List<ChangelingDna> changelingDnas = new List<ChangelingDna>();
+		private List<ChangelingMemories> changelingMemories = new List<ChangelingMemories>();
+		public List<ChangelingMemories> ChangelingMemories => changelingMemories;
```

Comment:
By doing this you are not really restricting the access to the list, I can still do ChangelingMemories.Clear() from an external class. To restrict the access you just have the list private and add public methods as the only interface to add, get and remove (if you need that) memories from the list.

Reply:
`public List<ChangelingMemories> ChangelingMemories => new(changelingMemories);`?

Reply:
No, that will create a new empty list with no connection to the real list.

What I meant in my commentary is to leave the list private and create a method to add memories

public void AddMemory(Changeling memory memory)

So you call that from the outside but you can't really break the content of the list.

I'm not entirely sure how the memories work or what they are used for so I can't give further details in my suggestion. I was just pointing out that what you were doing didn't really restrict the access to that list.

Reply:
It`s already in file....
And it`s actually creates a copy of list changelingMemories....

Reply:
Named AddMemories...